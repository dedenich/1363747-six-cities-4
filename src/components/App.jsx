import React from "react";
import Main from "./Main.jsx";

const App = (props) => {
  // eslint-disable-next-line react/prop-types
  const {offersCount} = props;
  return <React.Fragment>
    <Main
      offersCount={offersCount}
    />
  </React.Fragment>;
};

export default App;
