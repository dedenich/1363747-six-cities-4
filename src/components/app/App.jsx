import React, {PureComponent} from "react";
import PropTypes from 'prop-types';
import {Router, Route, Switch} from 'react-router-dom';
import {connect} from "react-redux";

import history from "../../history.js";
import {AppRoute} from "../../const.js";

import Main from "../main/Main.jsx";
import PropertyScreen from "../property-screen/property-screen.jsx";
import LoginScreen from "../login-screen/login-screen.jsx";

import withAuthForm from "../../hocs/with-auth-form/with-auth-form.jsx";

import {Operation as OfferOperation} from "../../reducers/offers/offers.js";
import {Operation as UserOperation} from "../../reducers/user/user.js";

import {getCurrentOffer} from "../../reducers/offers/selectors.js";

const LoginScreenWrapped = withAuthForm(LoginScreen);

class App extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      currentOffer: null,
    };

    this.handleHeadingClick = this.handleHeadingClick.bind(this);
  }

  handleHeadingClick(item) {
    return this.setState({currentOffer: item});
  }

  componentDidMount() {
    const {onLoadOffers, onCheckAuth} = this.props;
    onLoadOffers();
    onCheckAuth();
  }

  render() {

    return (
      <Router history={history}>
        <Switch>
          <Route path={AppRoute.ROOT} component={Main} exact />
          <Route path={AppRoute.SING_IN} component={LoginScreenWrapped} exact />
          <Route path={AppRoute.FAVORITES} exact />
          <Route path={AppRoute.ROOM} exact component={PropertyScreen}/>

        </Switch>
      </Router>
    );
  }
}

App.propTypes = {
  currentOffer: PropTypes.number,
  onLoadOffers: PropTypes.func,
  onCheckAuth: PropTypes.func,
  authorizationStatus: PropTypes.string,
};

const mapStateToProps = (state) => ({
  currentOffer: getCurrentOffer(state),
});

const mapDispatchToProps = (dispatch) => ({
  onLoadOffers() {
    dispatch(OfferOperation.loadOffers());
  },
  onCheckAuth() {
    dispatch(UserOperation.checkAuth());
  }
});

export {App};
export default connect(mapStateToProps, mapDispatchToProps)(App);
