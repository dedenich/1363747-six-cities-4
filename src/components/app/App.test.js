import React from 'react';
import App from './app';
import renderer from 'react-test-renderer';

it(`renders correctly`, () => {
  const tree = renderer
  .create(<App
    offersCount={123}
    captions={[
      `Description 1`,
      `Description 2`,
      `Description 3`,
      `Description 4`]}
  />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
