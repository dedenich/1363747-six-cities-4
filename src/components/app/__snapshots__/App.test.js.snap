// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="main.html"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height={41}
              src="img/logo.svg"
              width={81}
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://lh3.googleusercontent.com/proxy/w7U3brhOgXmvwNzrMYxTaxSKytM61krBAd9tYlzxDzJK8jH3yS5kGAv2xb5X248FV8fJ0aUY7ue3-wDqcI0hSxsUae29hRFsG8fXUeRbbSYkBPzLc7Q6hIyvt-Rw2uQGM3A0kwUDHNxNWhqCGXa4_bUk2EDzhw"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://static.wixstatic.com/media/f5a46b_351902e302894e0aafc6c871e9716fdf~mv2_d_4096_3066_s_4_2.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://66.media.tumblr.com/0fb91529c09818adb993f43de67bc5a2/d23535c215bd7d50-d8/s1280x1920/8d0c6bf1a57c0de78c9c1273c3909010b82807d9.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://i.pinimg.com/736x/39/a7/5a/39a75a834ef1aac8a4f36c30a3b1bbe1.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://lh3.googleusercontent.com/proxy/w7U3brhOgXmvwNzrMYxTaxSKytM61krBAd9tYlzxDzJK8jH3yS5kGAv2xb5X248FV8fJ0aUY7ue3-wDqcI0hSxsUae29hRFsG8fXUeRbbSYkBPzLc7Q6hIyvt-Rw2uQGM3A0kwUDHNxNWhqCGXa4_bUk2EDzhw"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://static.wixstatic.com/media/f5a46b_351902e302894e0aafc6c871e9716fdf~mv2_d_4096_3066_s_4_2.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              A Full Floor on Fifth Avenue With Central Park Views
            </h1>
            <button
              className="property__bookmark-button button"
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height={33}
                width={31}
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "80%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              3
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Apartment
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              2
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              5
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              120
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Wi-Fi
              </li>
              <li
                className="property__inside-item"
              >
                Washing machine
              </li>
              <li
                className="property__inside-item"
              >
                Towels
              </li>
              <li
                className="property__inside-item"
              >
                Heating
              </li>
              <li
                className="property__inside-item"
              >
                Coffee machine
              </li>
              <li
                className="property__inside-item"
              >
                Baby seat
              </li>
              <li
                className="property__inside-item"
              >
                Kitchen
              </li>
              <li
                className="property__inside-item"
              >
                Dishwasher
              </li>
              <li
                className="property__inside-item"
              >
                Cabel TV
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <style>
                #HostAvatar:after {display: none}
              </style>
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
                id="HostAvatar"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height={74}
                  src="https://images.unsplash.com/photo-1508214751196-bcfd4ca60f91?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
                  width={74}
                />
              </div>
              <span
                className="property__user-name"
              >
                Hanna
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                While we won’t be traveling to our local yoga studio anytime soon, we’re all for creating mini sanctuaries in our own home where we can practice restorative meditation and yoga. Our Design Milk Shop now carries Yoga Zeal which makes design-forward yoga mats made from a natural, biodegradable rubber. Paired with some of our other favorite products, you can create a space that you look forward to entering for a mindful break for deep breaths and stretches.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                1
              </span>
            </h2>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height={54}
                      src="img/avatar-max.jpg"
                      width={54}
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Max
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2019-04-24"
                  >
                    April 2019
                  </time>
                </div>
              </li>
            </ul>
            <form
              action="#"
              className="reviews__form form"
              method="post"
            >
              <label
                className="reviews__label form__label"
                htmlFor="review"
              >
                Your review
              </label>
              <div
                className="reviews__rating-form form__rating"
              >
                <input
                  className="form__rating-input visually-hidden"
                  defaultValue={5}
                  id="5-stars"
                  name="rating"
                  type="radio"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="5-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height={33}
                    width={37}
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  defaultValue={4}
                  id="4-stars"
                  name="rating"
                  type="radio"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="4-stars"
                  title="good"
                >
                  <svg
                    className="form__star-image"
                    height={33}
                    width={37}
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  defaultValue={3}
                  id="3-stars"
                  name="rating"
                  type="radio"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="3-stars"
                  title="not bad"
                >
                  <svg
                    className="form__star-image"
                    height={33}
                    width={37}
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  defaultValue={2}
                  id="2-stars"
                  name="rating"
                  type="radio"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="2-stars"
                  title="badly"
                >
                  <svg
                    className="form__star-image"
                    height={33}
                    width={37}
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  defaultValue={1}
                  id="1-star"
                  name="rating"
                  type="radio"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="1-star"
                  title="terribly"
                >
                  <svg
                    className="form__star-image"
                    height={33}
                    width={37}
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                className="reviews__textarea form__textarea"
                defaultValue=""
                id="review"
                name="review"
                placeholder="Tell how was your stay, what you like and what can be improved"
              />
              <div
                className="reviews__button-wrapper"
              >
                <p
                  className="reviews__help"
                >
                  To submit review please make sure to set 
                  <span
                    className="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least 
                  <b
                    className="reviews__text-amount"
                  >
                    50 characters
                  </b>
                  .
                </p>
                <button
                  className="reviews__submit form__submit button"
                  disabled={true}
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </section>
        </div>
      </div>
      <section
        className="property__map map"
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <article
            className="near-places__card place-card"
          >
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height={200}
                  src="img/room.jpg"
                  width={260}
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €80
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button place-card__bookmark-button--active button"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height={19}
                    width={18}
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    In bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                >
                  Wood and stone place
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Private room
              </p>
            </div>
          </article>
          <article
            className="near-places__card place-card"
          >
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height={200}
                  src="img/apartment-02.jpg"
                  width={260}
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €132
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height={19}
                    width={18}
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                >
                  Canal View Prinsengracht
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Apartment
              </p>
            </div>
          </article>
          <article
            className="near-places__card place-card"
          >
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height={200}
                  src="img/apartment-03.jpg"
                  width={260}
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €180
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height={19}
                    width={18}
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                >
                  Nice, cozy, warm big bed apartment
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Apartment
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>
</div>
`;
